### PR TITLE
OCLOMRS-5: Authentication Login UI implementation

### PR DESCRIPTION
**Pull request by hadijahkyampeire**
# JIRA TICKET NAME:
[OCLOMRS-5: Authentication Login UI implementation](https://issues.openmrs.org/browse/OCLOMRS-5)

# Summary:
This PR includes:
-  The login UI
- The login logic 
- And login tests

### CODE REVIEW

File: src/Actions/AuthActions.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  };
+};
+
+export {
+  AUTHENTICATED,
+  AUTHENTICATION_FAILED,
+  Login,
+  LoginFailed,
+  LoginAction
+};
```
Comment (emasys) [inline src/Actions/AuthActions.js]: Looks good, but the **action types**, **action creators** and **async action creators could be been in different files**, you can imagine if you had _10_ action types and more async action creators.

**Comment by mwaz** [comment]
kindly add blank lines to the end of your content on files

File: src/Actions/AuthActions.js (hunk 1 above)
Comment (hadijahkyampeire) [reply src/Actions/AuthActions.js]: Thanks @mwaz @emasys I am going to do that

File: src/__tests__/AuthReducers.test.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+});
+
+describe("AuthReducer", () => {
+  it("should not change state if no action passed", () => {
+    const state = {};
+    const action = {};
+    expect(authReducer(state, action)).toBe(state);
+  });
+
+  it("should respond coorectly to auth failure", () => {
```
[the remark this answers is not in the document]
Comment (hadijahkyampeire) [reply src/__tests__/AuthReducers.test.js]: Thanks @shakirandagire I have resolved the comments

File: src/Actions/AuthActions.js (hunk 1 above)
Comment (hadijahkyampeire) [reply src/Actions/AuthActions.js]: @emasys I have resolved your feedback, thanks

**Comment by EleisonC** [comment]
Please squash your commits

File: .gitignore [hunk 3]
```diff
@@ -21,4 +21,4 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 public/index.html~HEAD
-package-lock.json
```
Comment (flacode) [inline .gitignore]: Why did you remove the package-lock.json from .gitignore? This file changes for almost everyone and people will have to deal with many merge conflicts and a large diff file like [below](https://github.com/openmrs/openmrs-ocl-client/pull/2/files#diff-32607347f8126e6534ebc7ebaec4853d)

Comment (flacode) [inline .gitignore]: Why did you remove the package-lock.json from .gitignore? This file changes for almost everyone and people will have to deal with many merge conflicts and a large diff file like [below](https://github.com/openmrs/openmrs-ocl-client/pull/2/files#diff-32607347f8126e6534ebc7ebaec4853d)

File: src/ActionCreators/AuthActionCreators.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// Action type
```
Comment (flacode) [inline src/ActionCreators/AuthActionCreators.js]: Just a suggestion, its a common convention to have all actionTypes defined in one file. In the end, all actionTypes for the whole project will have a single point of reference.

File: .gitignore (hunk 3 above)
Comment (hadijahkyampeire) [reply .gitignore]: @flacode I removed it because it is the best practice given the kind of installation we did, It is helpful in case someone clones this repo and has to install the dependencies. It has the exact versions which avoids errors for anyone setting up the project. Refer to this for more info. [Why package-lock](https://docs.npmjs.com/files/package-lock.json)

Comment (flacode) [reply .gitignore]: @hadijahkyampeire, the package.json file keeps the versions of these dependencies. The package-lock.json file is an auto generated file that you have no control of how it changes. Therefore, there is no need of resolving merge conflicts in an auto generated file.

Comment (flacode) [reply .gitignore]: @hadijahkyampeire, the package.json file keeps the versions of these dependencies. The package-lock.json file is an auto generated file that you have no control of how it changes. Therefore, there is no need of resolving merge conflicts in an auto generated file.
